frame/contracts/src/lib.rs

```diff
@@ -946,6 +948,35 @@ struct InstantiateInput<T: Config> {
 	salt: Vec<u8>,
 }
 
+/// Determines whether events should be collected during execution.
+#[derive(PartialEq)]
+pub enum CollectEvents {
+	/// Collect events.
+	///
+	/// # Note
+	///
+	/// Events should only be collected when called off-chain, as this would
+	/// collect all the Events emitted in the block so far and put them in them into the PoV.
```

Comment:
```suggestion
	/// Events should only be collected when called off-chain, as this would otherwise
	/// collect all the Events emitted in the block so far and put them in them into the PoV.
```